### ISSUE TITLE
Translation does not fall back to English when the user's language is not covered

### ISSUE THREAD

**Issue by Chaos02** [mathieu-mp/homeassistant-intex-spa#53]
## Configuration

```yaml
default_config:

# various scripts and automation imports, all working
```

## Describe the bug

I see you've put [effort into error codes and their translations](https://github.com/mathieu-mp/homeassistant-intex-spa/blob/430a34db7051d97ae4e401ff22ad1ab10e2e8d36/custom_components/intex_spa/translations/sensor.en.json) but in home assistant the states of the error entities actually are just the lower case codes for me.

In case it wasn't thought of before I'd also appreciate it falling back to English, if a translation of your language isn't available yet.

(My HA instance is set to Germany, English)

I will provide a German translation pull request shortly!

## Debug log

```text

Logs following soon but probably not needed

```

**Comment by mathieu-mp** [comment]
Hi @Chaos02,

Thank you again for providing a German translation !

I'd also expect the fallback to English to work as-is.
It looks like the current translations implementation is deprecated for this integration.

This issue will require some work, but it has to be done !

**Comment by mathieu-mp** [comment]
Fixed by #86
